src/lib.rs

```diff
@@ -247,6 +313,26 @@ mod tests {
         assert!(ulid1 < ulid2);
     }
 
+    #[test]
+    fn test_order_monotonic() {
+        let dt = Utc::now();
+        let mut gen = Generator::new();
+        let ulid1 = gen.generate_from_datetime(dt).unwrap();
+        let ulid2 = gen.generate_from_datetime(dt).unwrap();
+        let ulid3 = Ulid::from_datetime(dt + Duration::milliseconds(1));
+        assert!(ulid1 < ulid2);
+        assert!(ulid2 < ulid3);
+        assert!(ulid2.timestamp_ms() < ulid3.timestamp_ms())
+    }
+
+    #[test]
+    fn test_increment() {
+        let ulid = Ulid::from_string("01BX5ZZKBKZZZZZZZZZZZZZZZX").unwrap();
+        let ulid = ulid.increment().unwrap(); // 01BX5ZZKBKZZZZZZZZZZZZZZZY
+        let ulid = ulid.increment().unwrap(); // 01BX5ZZKBKZZZZZZZZZZZZZZZZ
+        assert!(ulid.increment().is_none());
```

Comment:
Please add an additional test for overflow:
```
        let ulid = Ulid(u128::max_value());
        let ulid = ulid.increment();
```

Reply:
Thanks for being so thorough in your review. I did not know that rust would throw an exception in debug mode for int overflow. the more you know  🎉 :)